terraform/applications.tf

```diff
@@ -23,12 +23,24 @@ module "k8s" {
 module "k8s_worker" {
   source      = "git::https://github.com/canonical/k8s-operator//charms/worker/terraform"
   app_name    = module.k8s_worker_config.config.app_name
-  channel     = module.k8s_worker_config.config.channel 
+  channel     = coalesce(module.k8s_worker_config.config.channel, module.k8s_config.config.channel)
   config      = module.k8s_worker_config.config.config
   constraints = module.k8s_worker_config.config.constraints 
-  model  = var.model
+  model       = var.model
   resources   = module.k8s_worker_config.config.resources
   revision    = module.k8s_worker_config.config.revision
   base        = module.k8s_worker_config.config.base
   units       = module.k8s_worker_config.config.units
 }
+
+module "openstack" {
+  count           = var.cloud_integration == "openstack" ? 1 : 0
```

Comment:
if the cloud_integration variable is openstack, lets include that submodule

Reply:
man the modules really make it easy to do this conditional check

nice